scripts/bazel_diff.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+
+"""Computes the potentially differing rules from some git commit.
+
+Computes the rules that differ between a provided baseline commit and the
+current git commit.
+"""
+
+__copyright__ = """
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+
+import subprocess
+import argparse
+import tempfile
+import os
+import sys
+
+import scripts_utils
+
+
+def log(s: str) -> None:
+    print(s, file=sys.stderr)
+
+
+def make_bazel_diff_script(bazel: str, tmpdir: str) -> str:
```

Comment:
Consider using the pathlib.Path API instead of str and os.path.join. `Path` has a more modern interface, distinct type from `str`, and is generally accepted interchangeably by builtin APIs.